client/tx_generator.go

```diff
@@ -3,49 +3,41 @@ package client
 import (
 	"github.com/tendermint/tendermint/crypto"
 
-	"github.com/cosmos/cosmos-sdk/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/cosmos/cosmos-sdk/types/tx"
 )
 
 type (
 	// TxGenerator defines an interface a client can utilize to generate an
 	// application-defined concrete transaction type. The type returned must
 	// implement TxBuilder.
 	TxGenerator interface {
-		NewTx() TxBuilder
-		NewFee() Fee
-		NewSignature() Signature
-		MarshalTx(tx types.Tx) ([]byte, error)
-	}
-
-	Fee interface {
-		types.Fee
-		SetGas(uint64)
-		SetAmount(types.Coins)
-	}
-
-	Signature interface {
-		types.Signature
-		SetPubKey(crypto.PubKey) error
-		SetSignature([]byte)
+		NewTxBuilder() TxBuilder
+		// WrapTxBuilder wraps an existing tx in a TxBuilder or returns an error
+		WrapTxBuilder(tx sdk.Tx) (TxBuilder, error)
```

Comment:
@sahith-narahari I think you may benefit from having this on the `TxGenerator` interface for #6717

Reply:
yes, I need this for wrapping the migration. Thanks